File: versioned_docs/version-1.21.x/examples/js/src/data_ingestion_batch.mjs
import { Client } from "reduct-js";

// Create a client instance, then get or create a bucket
const client = new Client("http://127.0.0.1:8383", { apiToken: "my-token" });
const bucket = await client.getOrCreateBucket("my-bucket");

// Prepare a batch of records for different entries
const batch = bucket.beginWriteRecordBatch();
const timestamp = BigInt(Date.now()) * 1000n; // Current timestamp in microseconds
batch.add("sensor-1", timestamp, "Temperature: 20.5", "text/plain", {
  unit: "C",
});
batch.add("sensor-1", timestamp + 1000000n, "Temperature: 20.6", "text/plain", {
  unit: "C",
});
batch.add("camera-1", timestamp + 2000000n, "Frame #1", "text/plain");

// Write records from multiple entries in one request
const errors = await batch.send();

// Check for errors and throw first one
for (const entryErrors of errors.values()) {
  for (const [_timestamp, err] of entryErrors.entries()) {
    throw err;
  }
}

File: versioned_docs/version-1.21.x/examples/cpp/src/data_ingestion_batch.cc
#include <reduct/client.h>
#include <iostream>
#include <cassert>

using reduct::IBucket;
using reduct::IClient;
using reduct::Error;

using std::chrono_literals::operator ""s;

int main() {
    // Create a client instance, then get or create a bucket
    auto client = IClient::Build("http://127.0.0.1:8383", {.api_token="my-token"});
    auto [bucket, create_err] = client->GetOrCreateBucket("my-bucket");
    assert(create_err == Error::kOk);


    // Prepare a batch of records for different entries
    auto [record_errors, http_err] = bucket->WriteBatch([](auto batch) {
        IBucket::Time ts = IBucket::Time::clock::now();
        batch->AddRecord("sensor-1", ts, "Temperature: 20.5", "text/plain", {{"unit", "C"}});
        batch->AddRecord("sensor-1", ts + 1s, "Temperature: 20.6", "text/plain", {{"unit", "C"}});
        batch->AddRecord("camera-1", ts + 2s, "Frame #1", "text/plain");
    });
    assert(http_err == Error::kOk);

    // Check if all records were sent successfully
    for (auto& [entry, entry_errors] : record_errors) {
        for (auto& [ts, err] : entry_errors) {
            assert(err == Error::kOk);
        }
    }

    return 0;
}
